### PR TITLE
examples/dsc: add card detection via FS automount driver

### PR DESCRIPTION
**Pull request by pkarashchenko**

### CODE REVIEW

File: examples/dsc/dsc_file.c [hunk 1]
```diff
@@ -156,20 +156,19 @@ bool check_sdcard_content(void)
...
+ * Name: file_finalize()
  *
  * Description:
  *   Finalize file utility
  ****************************************************************************/
 
-void futil_finalize(void)
+void file_finalize(void)
 {
   sem_destroy(&g_sem);
```
Comment (pkarashchenko) [inline examples/dsc/dsc_file.c:167]: I'm not sure if now `g_sem` is needed at all with FS automounter.

**Comment by SPRESENSE** [comment]
@pkarashchenko Thanks for your contribution.
I have been trying out your changes https://github.com/apache/nuttx/pull/7584 and have noticed two things.

$ . /tools/configure.sh spresense:nsh_automount

1. The startup seems slower until the nsh prompt appears.
Fast boot time is an important performance feature for Spresense, 
Do you have the same symptoms on your side?

2. SD card mount fails.
First, turn on the power without an SD card inserted.
Next, insert the SD card. After that, it can not mount properly.
When the SD card is removed and inserted again, it is mounted correctly.

Could you please check these issues?
If needed, I will submit these issues to apache/nuttx?

**Comment by pkarashchenko** [comment]
Sure. Thank you for checking. I will take a look. One question from my side is how did you measure the time till the NSH prompt appears? Can that be easily noticed visually? Or you instrumented code and added some timestamp prints?

**Comment by pkarashchenko** [comment]
@SPRESENSE also one question from my side: is it slower when there is no card during bootup or it does not depend if the card is inserted during boot or not?

**Comment by pkarashchenko** [comment]
I was able to reproduce and fix both issues. My understanding of spresense PIO level was messed up so I spend few hours of educating myself via browsing the source code :)

**Comment by pkarashchenko** [comment]
@SPRESENSE please try the new PR to NuttX mainline. I hope issues to be resolved. Anyway those improvements are mostly for the case when board boots with no SD card. When SD card is inserted or is in the slot during the boot process it still needs around a second to settle down and set `Card State Stable` bit.

**Comment by SPRESENSE** [comment]
@pkarashchenko  san
I could confirm that these issues I reported before have been fixed. Thank you.
Could you please change the target branch to `develop` from `master`?
(Refer to [CONTRIBUTING.md](https://github.com/sonydevworld/spresense/blob/master/CONTRIBUTING.md).)

**Comment by pkarashchenko** [comment]
Done

**Comment by SPRESENSE** [comment]
@pkarashchenko san

I have checked this application with the latest NuttX, but it doesn't work well.

Regardless of whether CONFIG_FS_AUTOMOUNTER_DRIVER is enabled or not, the LCD display freezes when the SD card is removed or inserted.

After investigating the cause, I found that if I revert the following commits, it works correctly.

https://github.com/apache/nuttx/pull/7787/commits/a81dd93c19f9ecb61f6b7c2f32683f362c108947

I will merge it because they have nothing to do with this PR.

Thanks.

**Review by SPRESENSE — APPROVED** [review]

**Comment by pkarashchenko** [comment]
@SPRESENSE thank you. I made a test as you pointed out by copy example app and running it on top of example camera. But maybe it was before the PR that you specify. I added print in the place where automount signal is received by the app and it was working as expected. Let's follow-up to fix the other regression.

**Comment by pkarashchenko** [comment]
@SPRESENSE are there any other apps that should be updated? I'm think of the path to get board callback removed from the NuttX repo. I think that some defconfigs should be updated to use automount now. I would appreciate if you can give me some high level overview on what are next steps.

**Pull request merged**